prow/pj-on-kind.sh

```diff
@@ -25,29 +25,34 @@ function main() {
 
   # Generate PJ and Pod.
   mkpj "--config-path=${config}" "${job_config}" "--job=${job}" > "${PWD}/pj.yaml"
-  mkpod --build-id=snowflake "--prow-job=${PWD}/pj.yaml" --local "--out-dir=/output/${job}" > "${PWD}/pod.yaml"
+  mkpod --build-id=snowflake "--prow-job=${PWD}/pj.yaml" --local "--out-dir=${out_dir}" > "${PWD}/pod.yaml"
+
+  # Add any k8s resources that the pod depends on to the kind cluster here. (secrets, configmaps, etc.)
 
   # Deploy pod and watch.
   echo "Applying pod to the mkpod cluster. Configure kubectl for the mkpod cluster with:"
-  echo '>  export KUBECONFIG="$(kind get kubeconfig-path --name="mkpod")"'
+  echo '>  export KUBECONFIG="$(kind get kubeconfig-path --name=mkpod)"'
   pod=$(kubectl apply -f "${PWD}/pod.yaml" | cut -d ' ' -f 1)
   kubectl get "${pod}" -w
 }
 
 # Prep and check args.
 function parseArgs() {
-  job="${1:-""}"
-  config="${CONFIG_PATH:-""}"
-  job_config="${JOB_CONFIG_PATH:-""}"
-  out_dir="${OUT_DIR:-"/tmp/prowjob-out"}"
-  kind_config="${KIND_CONFIG:-""}"
+  # Use node mounts under /etc/ so pods behave well on COS nodes too. https://cloud.google.com/container-optimized-os/docs/concepts/disks-and-filesystem
+  job="${1:-}"
+  config="${CONFIG_PATH:-}"
+  job_config="${JOB_CONFIG_PATH:-}"
+  out_dir="${OUT_DIR:-/etc/prowjob-out/${job}}"
+  kind_config="${KIND_CONFIG:-}"
+  node_dir="${NODE_DIR:-/etc/kind-node}"  # Any pod hostPath mounts should be under this dir to reach the true host via the kind node.
```

Comment:
nit: this is sort of a mis-use of `/etc`, this should probably under `/var/`, `/run` or `/mnt`?
https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard
./shrug

Reply:
I was going off of https://cloud.google.com/container-optimized-os/docs/concepts/disks-and-filesystem

Looks like I can use `/mnt/disks/kind-node` instead.  Should I change `/etc/prowjob-out` to `/mnt/disks/prowjob-out` as well?

Reply:
So inside the kind node you should be able to do any path, I would do one of the listed ones or invent your own /prow just because these are fairly standardized.

On the host machine I'd think you'd write to something in Home owned by the user. /etc is for system config.

Why is this running on COS directly?

Reply:
> Why is this running on COS directly?

I want the generated pod.yaml files to be suitable for use in a real cluster in addition to the kind cluster. I tried to use a path like `/prow`, but COS doesn't allow that.